packet/routes/api.py

```diff
@@ -20,25 +24,53 @@ def sign(packet_id, info):
             for sig in filter(lambda sig: sig.member == info["uid"], packet.upper_signatures):
                 sig.signed = True
                 app.logger.info("Member {} signed packet {} as an upperclassman".format(info["uid"], packet_id))
+                packet_signed_notification(packet, info["uid"])
                 return commit_sig(packet, was_100)
 
             # The CSHer is a misc so add a new row
             db.session.add(MiscSignature(packet=packet, member=info["uid"]))
             app.logger.info("Member {} signed packet {} as a misc".format(info["uid"], packet_id))
+            packet_signed_notification(packet, info["uid"])
             return commit_sig(packet, was_100)
         else:
             # Check if the freshman is onfloor and if so, sign that row
             for sig in filter(lambda sig: sig.freshman_username == info["uid"], packet.fresh_signatures):
                 sig.signed = True
                 app.logger.info("Freshman {} signed packet {}".format(info["uid"], packet_id))
+                packet_signed_notification(packet, info["uid"])
                 return commit_sig(packet, was_100)
 
     app.logger.warn("Failed to add {}'s signature to packet {}".format(info["uid"], packet_id))
     return "Error: Signature not valid.  Reason: Unknown"
 
+
+@app.route("/api/v1/subscribe/", methods=["POST"])
+@packet_auth
+@before_request
+def subscribe(info):
+    data = request.form
+    if app.config["REALM"] == "csh":
+        subscription = NotificationSubscription(token=data['token'], member=info["uid"])
```

Comment:
_Joel is a broken record._

Reply:
What do you mean?

Reply:
I was talking about the quotes in this file; sorry.